packages/interactive_media_ads/lib/src/ad.dart

```diff
@@ -0,0 +1,199 @@
+
+  /// The playback time before the ad becomes skippable.
+  ///
+  /// The value is null for non-skippable ads, or if the value is unavailable.
+  Duration? get skipTimeOffset => platform.skipTimeOffset;
+
+  /// The URL associated with the survey for the given ad.
+  String? get surveyUrl => platform.surveyUrl;
+
+  /// The title of this ad from the VAST response.
+  String? get title => platform.title;
+
+  /// The custom parameters associated with the ad at the time of ad
+  /// trafficking.
+  String get traffickingParameters => platform.traffickingParameters;
+
+  /// The set of ad UI elements rendered by the IMA SDK for this ad.
+  Set<AdUIElement> get uiElements => platform.uiElements;
+
+  /// The list of all universal ad IDs for this ad.
+  List<UniversalAdId> get universalAdIds => List<UniversalAdId>.unmodifiable(
+    platform.universalAdIds.map(UniversalAdId.fromPlatform),
+  );
+
+  /// The VAST bitrate in Kbps of the selected creative.
+  int get vastMediaBitrate => platform.vastMediaBitrate;
+
+  /// The VAST media height in pixels of the selected creative.
+  int get vastMediaHeight => platform.vastMediaHeight;
+
+  /// The VAST media width in pixels of the selected creative.
+  int get vastMediaWidth => platform.vastMediaWidth;
+
+  /// The width of the selected creative if non-linear, else returns 0.
+  int get width => platform.width;
+
+  /// Indicates whether the ad’s current mode of operation is linear or
+  /// non-linear.
+  bool get isLinear => platform.isLinear;
+
+  /// Indicates whether the ad can be skipped by the user.
+  bool get isSkippable => platform.isSkippable;
+}
+
+/// Simple data object containing podding metadata.
+class AdPodInfo {
+  /// Constructs an [AdPodInfo] from a specific platform implementation.
+  AdPodInfo.fromPlatform(this.platform);
+
+  /// Implementation of [PlatformAdPodInfo] for the current platform.
+  final PlatformAdPodInfo platform;
+
+  /// The position of the ad within the pod.
+  ///
+  /// The value returned is one-based, for example, 1 of 2, 2 of 2, etc. If the
+  /// ad is not part of a pod, this will return 1.
+  int get adPosition => platform.adPosition;
+
+  /// The maximum duration of the pod.
+  ///
+  /// For unknown duration, null.
+  Duration? get maxDuration => platform.maxDuration;
+
+  /// Returns the index of the ad pod.
+  ///
+  /// Client side: For a preroll pod, returns 0. For midrolls, returns 1, 2,…,
+  /// N. For a postroll pod, returns -1. Defaults to 0 if this ad is not part of
+  /// a pod, or this pod is not part of a playlist.
+  ///
+  /// DAI VOD: Returns the index of the ad pod. For a preroll pod, returns 0.
+  /// For midrolls, returns 1, 2,…,N. For a postroll pod, returns N+1…N+X.
+  /// Defaults to 0 if this ad is not part of a pod, or this pod is not part of
+  /// a playlist.
+  ///
+  /// DAI live stream: For a preroll pod, returns 0. For midrolls, returns the
+  /// break ID. Returns -2 if pod index cannot be determined (internal error).
```

Comment:
Can we make constants for these magic values, and document those instead of the raw values?

Reply:
Hmm I don't know how to reconcile that this value can be any positive integer when it's a mid rolls. Also, postroll are represented differently depending if you are using client side ads or DAI VOD and there isn't a well to get the type in the class.